Foundation/NSString.swift

```diff
@@ -1301,12 +1301,18 @@ extension NSString {
         let readResult = try NSData(contentsOf: url, options:[])
 
         let bytePtr = readResult.bytes.bindMemory(to: UInt8.self, capacity:readResult.length)
-        if readResult.length >= 2 && bytePtr[0] == 254 && bytePtr[1] == 255 {
+        if readResult.length >= 4 && bytePtr[0] == 255 && bytePtr[1] == 254 && bytePtr[2] == 0 && bytePtr[3] == 0 {
```

Comment:
might be good to use the hex values for these since most sources of the unicode standards refer to 0xFF and 0xFE etc. (this is just a style nit not a functionality commentary)